var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -51,9 +52,22 @@ class Charliecloud(AutotoolsPackage):
     def configure_args(self):
 
         args = []
+        py_path = '{0}'.format(self.spec['python'].prefix.bin.join('python3'))
```

Comment:
```suggestion
        py_path = self.spec['python'].command.path
```
This will work for both Python 2 and 3.

Reply:
Any suggestions on how to do something similar with the `sphinx-build` executable? The `py-sphinx` package name doesn't match the executable name `sphinx-build`, so I run into an error.

```
+            sphinx_path = self.spec['py-sphinx'].command.path
```
```
==> Error: RuntimeError: Unable to locate py-sphinx command in /var/tmp/jogas/spack_packages/linux-centos7-haswell/gcc-4.8.5/py-sphinx-3.0.0-rkxltbqbqesiywu5nzu6zrga6xflxyuh/bin
```

Reply:
It's fine to do things the way you're already doing since the executable name doesn't depend on the version, or you could override the `command` property of the package if you want to.